app/assets/javascripts/textbook_transactions.js

```diff
@@ -0,0 +1,428 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
+// You can use CoffeeScript in this file: http://coffeescript.org/
+
+var listingsData = [],
+	booksData = [];
+
+var ready = function() {
+
+	var listingsToShow,
+		listingList,
+		booksToShow,
+		bookList;
+
+	const default_book_cover = '/assets/icons/no_book.png'
+
+	// MyListings page, actions (remove, renew, report)
+	if ($('#my-listing-table').length) {
+		$(document).on('click', '.action', function() {
+			var actionName = $(this).attr('action'),
+				listing_id = $(this).attr('id');
+
+			if (actionName == "report") {
+				$('#report-bug-modal').modal();
+			} else {
+				$.ajax({
+					url: '/textbook_transactions/' + actionName,
+					dataType: 'json',
+					type: 'POST',
+					data: {
+						listing_id: listing_id
+					},
+					success: function(data) {
+						location.reload();
+					},
+					error: function(data) {
+						alert("Something went wrong :(");
+					}
+				});
+			}
+
+		});
+	}
+
+	// Infinite Scroll
+	$('#main-container').scroll(function() {
+		if ($('#main-container').prop('scrollHeight') - $('#main-container').scrollTop() <= $('#main-container').height() + 100) {
+			if ($('#book-titles').length) {
+				window.setTimeout(appendBooks, 500);
+			}
+			if ($('#listing-titles').length) {
+				window.setTimeout(appendListings, 500);
+			}
+		}
+	});
+
+	// Load textbook data
+	if ($('#post-listing').length) {
+		$.ajax({
+			url: '/textbooks',
+			dataType: 'json',
+			type: 'GET',
+			success: function(data) {
+				// data is an array of objects
+				// Each object has the attributes: id, title, medium_image_link, mnemonic_numbers, follower_count
+				booksData = data;
+				if ($('#book-titles').length) {
+					displayBooks(booksData);
+				}
+			}
+		});
+	}
+
+	// Load listings data
+	if ($('#listing-titles').length) {
+		$.ajax({
+			url: '/textbook_transactions/listings',
+			dataType: 'json',
+			type: 'GET',
+			success: function(data) {
+				$('#listing-titles').removeAttr('disabled');
+				$('#listing-titles').attr('placeholder','e.g. ECON 2010 or Fatal Equilibrium');
+				listingsData = data;
+				displayListings(listingsData);
+			}
+		});
+	}
+
+	// Book show page - Load relevant listings data
+	if ($('#follow').length) {
+		var id = $('#book-id').text().trim();
+
+		$.ajax({
+			url: '/textbook_transactions/listings' + "?book_id=" + id,
```

Comment:
use the data: {} and let jquery handle parameter formatting instead of raw "?" - you might not URI escape (unlikely but be safe)